packaging/os/rudix.py

```diff
@@ -0,0 +1,149 @@
+author: James Polley
+short_description: Package manager for Rudix
+description:
+    - Manages Rudix packages
+version_added: "1.9.2"
+options:
+    name:
+        description:
+            - name of package to install/remove
+        required: true
+    state:
+        description:
+            - state of the package
+        choices: [ 'present', 'absent', 'active', 'inactive' ]
+        required: false
+        default: present
+    update_cache:
+        description:
+            - update the package db first
+        required: false
+        default: "no"
+        choices: [ "yes", "no" ]
+notes:  []
+'''
+EXAMPLES = '''
+- rudix: name=foo state=present
+- rudix: name=foo state=present update_cache=yes
+- rudix: name=foo state=absent
+- rudix: name=foo state=active
+- rudix: name=foo state=inactive
+'''
+
+import pipes
+
+def update_package_db(module, rudix_path):
+    """ Updates packages list. """
+    pass #Package list is always downloaded fresh
+
+
+def query_package(module, rudix_path, name, state="present"):
+    """ Returns whether a package is installed or not. """
+
+    if state == "present":
+
+        rc, out, err = module.run_command("%s --list | grep -q ^.*%s" % (pipes.quote(rudix_path), pipes.quote(name)), use_unsafe_shell=True)
+        if rc == 0:
+            return True
+
+        return False
+
+
+def remove_packages(module, rudix_path, packages):
+    """ Uninstalls one or more packages if installed. """
+
+    remove_c = 0
+    # Using a for loop incase of error, we can report the package that failed
+    for package in packages:
+        # Query the package first, to see if we even need to remove
+        if not query_package(module, rudix_path, package):
+            continue
+
+        rc, out, err = module.run_command("%s --remove %s" % (rudix_path, package))
+
+        if query_package(module, rudix_path, package):
+            module.fail_json(msg="failed to remove %s: %s" % (package, out))
+
+        remove_c += 1
+
+    if remove_c > 0:
+
+        module.exit_json(changed=True, msg="removed %s package(s)" % remove_c)
+
+    module.exit_json(changed=False, msg="package(s) already absent")
+
+
+def install_packages(module, rudix_path, packages):
+    """ Installs one or more packages if not already installed. """
+
+    install_c = 0
+
+    for package in packages:
+        if query_package(module, rudix_path, package):
+            continue
+
+        rc, out, err = module.run_command("%s --install %s" % (rudix_path, package))
+
+        if not query_package(module, rudix_path, package):
+            module.fail_json(msg="failed to install %s: %s" % (package, out))
+
+        install_c += 1
+
+    if install_c > 0:
+        module.exit_json(changed=True, msg="installed %s package(s)" % (install_c))
+
+    module.exit_json(changed=False, msg="package(s) already present")
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            name = dict(aliases=["pkg"], required=True),
```

Comment:
make name a list, then you should be able to avoid the split and it will take both a list or , separated string as arguments.